### PR TITLE
xfce: does not explicitly require a gvfs package

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

- In order to use GIO/GVFS it is enough to enable the `gvfs` service.

- The module option `services.gvfs.package` can be used to choose a  variation of the `gvfs` package, if desired.

Related to https://github.com/NixOS/nixpkgs/issues/69208.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121046` at 54247e98 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>xfce.gigolo</li>
    <li>xfce.thunar (xfce.thunar-bare ,xfce.thunar-build ,xfce.thunarx-2-dev)</li>
    <li>xfce.thunar-archive-plugin</li>
    <li>xfce.thunar-dropbox-plugin</li>
    <li>xfce.xfdesktop</li>
  </ul>
</details>

File: nixos/doc/manual/release-notes/rl-2105.xml [hunk 1]
```diff
@@ -978,6 +978,24 @@ environment.systemPackages = [
...
+     <link xlink:href="https://docs.xfce.org/apps/gigolo/">gigolo</link>.
+     For that to work, the gvfs nixos service is enabled by default,
+     and it can be configured with the specific package that provides
+     GVfs. Until now Xfce4 was setting it to use a lighter version of
+     GVfs (without support for samba). To avoid conflicts with other
+     desktop environments this setting has been dropped. Users that
+     still want it should add the following to their system
+     configuration:
+     <programlisting>
+       <xref linkend="opt-services.gvfs.enable" /> = pkgs.gvfs.override { samba = null; };
```
Comment (symphorien) [inline nixos/doc/manual/release-notes/rl-2105.xml:995]: hum enable is a boolean option. Are you mixing with services.gvfs.package ?

**Review by symphorien — CHANGES_REQUESTED** [review]

File: nixos/doc/manual/release-notes/rl-2105.xml (hunk 1 above)
Comment (romildo) [reply nixos/doc/manual/release-notes/rl-2105.xml]: Thanks for pointing it out. Fixed.

File: nixos/doc/manual/release-notes/rl-2105.xml [hunk 2]
```diff
@@ -984,6 +984,24 @@ environment.systemPackages = [
...
+     <link xlink:href="https://docs.xfce.org/apps/gigolo/">gigolo</link>.
+     For that to work, the gvfs nixos service is enabled by default,
+     and it can be configured with the specific package that provides
+     GVfs. Until now Xfce4 was setting it to use a lighter version of
+     GVfs (without support for samba). To avoid conflicts with other
+     desktop environments this setting has been dropped. Users that
+     still want it should add the following to their system
+     configuration:
+     <programlisting>
+       <xref linkend="opt-services.gvfs.package" /> = pkgs.gvfs.override { samba = null; };
```
Comment (symphorien) [inline nixos/doc/manual/release-notes/rl-2105.xml:1001]: I think setting the enable option to true is missing.
```suggestion
       <xref linkend="opt-services.gvfs.enable" /> = true;
       /* The following is only necessary to use a slimmed down version of gvfs, without samba support */
       <xref linkend="opt-services.gvfs.package" /> = pkgs.gvfs.override { samba = null; };
```

**Review by symphorien — CHANGES_REQUESTED** [review]

File: nixos/doc/manual/release-notes/rl-2105.xml (hunk 2 above)
Comment (jtojnar) [reply nixos/doc/manual/release-notes/rl-2105.xml:1001]: Xfce enables it by default according to the note.

File: nixos/doc/manual/release-notes/rl-2105.xml [hunk 3]
```diff
@@ -984,6 +984,24 @@ environment.systemPackages = [
...
+     </programlisting>
```
Comment (jtojnar) [inline nixos/doc/manual/release-notes/rl-2105.xml:1002]: This should not be indented, though.

File: nixos/doc/manual/release-notes/rl-2105.xml (hunk 2 above)
Comment (romildo) [reply nixos/doc/manual/release-notes/rl-2105.xml:1001]: Yes, gvfs is enabled by default in the xfce module.

File: nixos/doc/manual/release-notes/rl-2105.xml (hunk 3 above)
Comment (romildo) [reply nixos/doc/manual/release-notes/rl-2105.xml]: Do you mean the content of `programlisting`?
Fixed.

There are several occurrences of `programlisting` elements with  indented content like that on the document, though.

File: nixos/doc/manual/release-notes/rl-2105.xml (hunk 2 above)
Comment (symphorien) [reply nixos/doc/manual/release-notes/rl-2105.xml]: Oh sorry I had misread. Please ignore.

**Review by symphorien — APPROVED** [review]

File: nixos/doc/manual/release-notes/rl-2105.xml (hunk 3 above)
Comment (jtojnar) [reply nixos/doc/manual/release-notes/rl-2105.xml]: Yeah, programlisting renders the spaces verbatim.

**Pull request merged**